Lib/unittest/test/testmock/testpatch.py

```diff
@@ -664,6 +664,22 @@ def test():
         test()
 
 
+    def test_patch_dict_decorator_resolution(self):
+        # bpo-35512: Ensure that patch with a string target resolves to
+        # the new dictionary during function call
+        original = support.target.copy()
+
+        @patch.dict('unittest.test.testmock.support.target', {'bar': 'BAR'})
+        def test():
+            self.assertEqual(support.target, {'foo': 'BAZ', 'bar': 'BAR'})
+
+        try:
+            support.target = {'foo': 'BAZ'}
+            test()
```

Comment:
You could always add the following here, so be super sure:
`self.assertEqual(support.target, {'foo': 'BAZ'})`

Reply:
Sure, added with d7f5c11162a5b3809d8a5864ea359e904e14f671